runelite-client/src/main/java/net/runelite/client/ui/overlay/worldMap/WorldMapOverlay.java

```diff
@@ -0,0 +1,136 @@
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.ui.overlay.worldMap;
+
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.Rectangle;
+import java.util.List;
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.Client;
+import net.runelite.api.Point;
+import net.runelite.api.widgets.Widget;
+import static net.runelite.api.widgets.WidgetID.WORLD_MAP;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.OverlayPriority;
+
+@Singleton
+@Slf4j
+public class WorldMapOverlay extends Overlay
+{
+	private final WorldMapPointManager worldMapPointManager;
+	private final Provider<Client> clientProvider;
+	private final WorldMapService worldMapService;
+	private final WorldMapOverlayMouseListener worldMapOverlayMouseListener;
+
+	@Inject
+	public WorldMapOverlay(Provider<Client> clientProvider, WorldMapPointManager worldMapPointManager, WorldMapService worldMapService, WorldMapOverlayMouseListener worldMapOverlayMouseListener)
+	{
+		this.clientProvider = clientProvider;
+		this.worldMapPointManager = worldMapPointManager;
+		this.worldMapService = worldMapService;
+		this.worldMapOverlayMouseListener = worldMapOverlayMouseListener;
+		setPosition(OverlayPosition.DYNAMIC);
+		setPriority(OverlayPriority.HIGH);
+		setLayer(OverlayLayer.ALWAYS_ON_TOP);
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		final List<WorldMapPoint> points = worldMapPointManager.getWorldMapPoints();
+
+		if (points.isEmpty())
+		{
+			return null;
+		}
+
+		final Client client = clientProvider.get();
+		final Rectangle worldMapRectangle = worldMapService.getMapDisplayRectangle();
+
+		Widget widget = client.getWidget(WORLD_MAP, 0);
+		if (widget == null)
+		{
+			return null;
+		}
+
+		for (WorldMapPoint worldPoint : points)
+		{
+			if (worldPoint.getImage() != null && worldPoint.getWorldPosition() != null)
```

Comment:
Inverse if, then it reads better

Reply:
❔ I don't see why, this is the normal pattern for a double null-check in Java code.